src/utils/array.js

```diff
@@ -0,0 +1,3 @@
+export const getIntersection = (array1, array2) => {
+  return array1.filter((value) => array2.includes(value));
```

Comment:
혹시 시간복잡도라고 들어보셨나요? 지금 사용해주신 건 filter에 includes를 중첩한 상태인데, 이건 for 문 두 개를 중첩한 것처럼 O(n²) 시간이 걸립니당
그럼 이건 어떻게 개선할 수 있을까요? (힌트 : `set.has(value)`)

Reply:
하리의 코멘트를 읽고 살펴보니 array1를 순회하면서 array2도 중첩해서 순회하는 형태였네요..!
array2을 `Set`으로 생성하고 array1의 요소 중에 set에 등록되어 있는 값이 존재하는지 `has` 메서드로 체크한다면 한 번의 순회로 개선할 수 있을 것 같습니다. 힌트 감사합니다 ㅎㅎ

Reply:
하리의 코멘트를 읽고 살펴보니 array1를 순회하면서 array2도 중첩해서 순회하는 형태였네요..!

```suggestion
export const getIntersection = (array1, array2) => {
  const arraySet = new Set(array2);
  return array1.filter((value) => arraySet.has(value));
```
이렇게 array2을 `Set`으로 생성하고 array1의 타겟 요소가 array2 set에 존재하는 값인지 `has` 메서드로 체크한다면 한 번의 순회로 개선할 수 있을 것 같습니다. 힌트 감사합니다 ㅎㅎ

Reply:
코멘트에 작성한 내용대로 유틸 함수 `getIntersection`을 리팩토링 하였습니다!

[refactor: 두 배열의 공통 요소를 찾는 유틸 함수의 로직 개선 (440718b)](https://github.com/woowacourse/javascript-lotto/pull/335/commits/440718b30a7f4ef31355c30c319ebe21f7be7913)